**PROJECT DESCRIPTION**

This project explores data related to Guillain-Barré Syndrome from two sources clinical research data and search engine results. Using PySpark, I perform data cleaning, transformation, and exploratory analysis to understand patterns and insights from each dataset independently.

**Fetching and Setting Up Data from Clinical and SerpApi APIs**

In [0]:
# Fetching Clinical Trials Data for Guillain-Barré Syndrome using Clinicaltrials.gov API

import requests

# URL and parameters
url = "https://clinicaltrials.gov/api/v2/studies"
params = {
    "query.cond": "Guillain-Barre",
    "pageSize": "100"
}

# To store all results
all_studies = []  

while True:
    # API request
    response = requests.get(url, params=params)
    data = response.json()
    
    # Append current page studies
    all_studies.extend(data.get("studies", []))
    
    # Check for next page
    next_token = data.get("nextPageToken")
    if not next_token:
        break
    
    # Set token for next page
    params["pageToken"] = next_token

# Store full JSON response
clinical_data = {"studies": all_studies}

# Display the results in JSON format
display(clinical_data)


{'studies': [{'protocolSection': {'identificationModule': {'nctId': 'NCT05324176',
     'orgStudyIdInfo': {'id': 'MD.19.06.186.R1.R2'},
     'organization': {'fullName': 'Mansoura University', 'class': 'OTHER'},
     'briefTitle': 'Diaphragm Thickness by Ultrasonography in Neurological Disorders',
     'officialTitle': 'Assessment of Diaphragm Thickness by Ultrasonography in Guillain Barre Syndrome and Myasthenia Gravis'},
    'statusModule': {'statusVerifiedDate': '2022-05',
     'overallStatus': 'COMPLETED',
     'expandedAccessInfo': {'hasExpandedAccess': False},
     'startDateStruct': {'date': '2022-04-13', 'type': 'ACTUAL'},
     'primaryCompletionDateStruct': {'date': '2022-05-09', 'type': 'ACTUAL'},
     'completionDateStruct': {'date': '2022-05-13', 'type': 'ACTUAL'},
     'studyFirstSubmitDate': '2022-04-04',
     'studyFirstSubmitQcDate': '2022-04-04',
     'studyFirstPostDateStruct': {'date': '2022-04-12', 'type': 'ACTUAL'},
     'lastUpdateSubmitDate': '2022-05-13',
     '

In [0]:
# Fetching Google Search Results for Guillain-Barré Syndrome using SerpApi

#%pip install google-search-results==2.4.2

from serpapi import GoogleSearch
import json

api_key = "1e9e3313e442d407a5bb0f40096ff8633cfd38190089299db0626007b6c08632"

# To store all results
fetched_results_raw = []

page_size = 10
max_pages = 10

# Loop over pages to fetch results
for page in range(max_pages):
    params = {
        "api_key": api_key,
        "engine": "google",
        "q": "Guillain Barre Syndrome",
        "location": "India",
        "google_domain": "google.co.in",
        "gl": "in",
        "hl": "hi",
        "cr": "countryUS",
        "start": page * page_size
    }

    search = GoogleSearch(params)
    results = search.get_dict()

    fetched_results_raw += results.get("organic_results", [])

# Display the results in JSON format
print(json.dumps(fetched_results_raw, indent=2, ensure_ascii=False))


[
  {
    "position": 1,
    "title": "Guillain-Barre syndrome - Symptoms and causes",
    "link": "https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793",
    "redirect_link": "https://www.google.co.in/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793&ved=2ahUKEwiLmrWzsMSNAxXQgFYBHfVIOPgQFnoECBsQAQ",
    "displayed_link": "https://www.mayoclinic.org › ...",
    "thumbnail": "https://serpapi.com/searches/6836119ef052758ead83f85a/images/0d6cc8c82e5038e13c997e42de84add251d713f8066057e7f1cc49f0332ac3e9.jpeg",
    "favicon": "https://serpapi.com/searches/6836119ef052758ead83f85a/images/0d6cc8c82e5038e13c997e42de84add2ca730e8d348b329fa7508194cb2eb761.png",
    "date": "7 जून 2024",
    "snippet": "Guillain-Barre syndrome often begins with tingling and weakness starting in the feet and legs and spreading to the upper body and arms. Some ...",
    "snippet

**Convert Clinical Trials and SerpApi JSON Data to Pandas DataFrames (Normalization)**

In [0]:
# Convert Clinical Trials JSON Data to Pandas DataFrame for Analysis

import requests
import pandas as pd

url = "https://clinicaltrials.gov/api/v2/studies"
params = {
    "query.cond": "Guillain-Barre",
    "pageSize": "100"  
}

response = requests.get(url, params=params)
data = response.json()

studies = data.get("studies", [])

# Creating an empty list to collect trial records
trials_list = []

# Loop through each study
for study in studies:
    try:
        id_module = study.get('protocolSection', {}).get('identificationModule', {})
        status_module = study.get('protocolSection', {}).get('statusModule', {})

        trials_list.append({
            "nctId": id_module.get('nctId', ''),
            "briefTitle": id_module.get('briefTitle', ''),
            "overallStatus": status_module.get('overallStatus', ''),
            "hasResults": study.get('hasResults', False)
        })

    except Exception as e:
        print(f"Error processing a study: {e}")

# Convert to DataFrame
df = pd.DataFrame(trials_list)

# Display the DataFrame
display(df.head(100))




nctId briefTitle overallStatus hasResults NCT05324176 Diaphragm Thickness by Ultrasonography in Neurological Disorders COMPLETED false NCT01655394 Change of Nerve Conduction Properties in IVIg Dependent Neuropathies UNKNOWN false NCT00271791 Prednisone Treatment for Vestibular Neuronitis COMPLETED false NCT06822231 High-Tech Rehabilitation Pathway for Acute Adult Neuromuscular Diseases - Fit4MedRob-Acute MND Project NOT_YET_RECRUITING false NCT06885762 Efgartigimod for the Treatment of Guillain-Barré Syndrome NOT_YET_RECRUITING false NCT05095844 National Vaccine Adverse Event Reporting Survey and Etiology UNKNOWN false NCT05701189 Evaluating Efgartigimod in Patients with Guillain-Barré Syndrome RECRUITING false NCT04752566 A Study to Evaluate the Efficacy and Safety of Eculizumab in Guillain-Barré Syndrome COMPLETED true NCT04829526 Firm Observational Clinical Unicenter Study on Guillain Barré Syndrome RECRUITING false NCT02493725 JET-GBS - Japanese Eculizumab Trial for GBS COMPLETED false NCT04166357 Early Prediction of Respiratory and Autonomic Complications of GBS Using Neuromuscular Ultrasound UNKNOWN false NCT05940922 RWE-based Treatment Patterns and Outcomes in CIDP NOT_YET_RECRUITING false NCT05212792 Genomics and COVID-19 Vaccine Adverse Events RECRUITING false NCT04303962 Efficacy of Intravenous Gamma Globulin on Guillain-Barre Syndrome UNKNOWN false NCT03943589 A Study of Imlifidase in Patients With Guillain-Barré Syndrome COMPLETED true NCT05104762 IVIG Versus Plasmapheresis in the Treatment of Guillian Barrie Syndrome Patients COMPLETED false NCT06615622 Our Study Aims to Determine If Nerve Alterations in Acute GBS and CIDP Detectable by Ultrasound Match Electrodiagnostic Findings and If This Method Aids Early Diagnosis, Predict Their Outcomes and Differentiate Between Axonal and Demyelinating Subtypes. RECRUITING false NCT05284786 Ultrasonography and Electrophysiology in GBS UNKNOWN false NCT01005524 Brain Computer Interface for Communication in ICU: a Feasibility Study COMPLETED false NCT06939634 Evaluation of Usability and Safety of the System Atalante in Patients With High Paraplegia and Tetraplegia COMPLETED false NCT04035135 A Clinical Study of ANX005 and IVIG in Subjects With Guillain Barré Syndrome (GBS) COMPLETED false NCT00167622 Early Mechanical Ventilation for Guillain Barré Syndrome COMPLETED false NCT01582763 International Guillain-Barré Syndrome Outcome Study COMPLETED false NCT03801135 Changes in Haemostasis After Therapeutic Plasmapheresis With Citrate Anticoagulation UNKNOWN false NCT03941184 Spontaneous Coronary Artery Dissection (SCAD) and Autoimmunity COMPLETED false NCT04092140 Neuromuscular Ultrasound in Correlation to Neurophysiological Findings UNKNOWN false NCT03268304 Feasibility of Two New Software Modules for the Rehabilitation of Patients With Neuromuscular Upper Limb Impairments WITHDRAWN false NCT00411216 Recovery of Visual Acuity in People With Vestibular Deficits COMPLETED true NCT01469858 Perception and Multisensory Integration in Neurological Patients Using fMRI UNKNOWN false NCT06167239 Ventilator Trigger Sensitivity Adjustment Versus Threshold Inspiratory Muscle Training on Arterial Blood Gases RECRUITING false NCT04674163 Expression Profile of ERK5 and PKM2 Kinases in Neuroinflammatory Diseases. WITHDRAWN false NCT05630313 Gene Sequencing as a Strategy for Identifying Genetic Factors Associated With Serious Adverse Events After Covid-19 Vaccines in Use in Brazil RECRUITING false NCT02459808 Gastrointestinal Transit Time in Patients With Guillain-Barré Syndrome WITHDRAWN false NCT06605612 Development and Validation of the FBIndex to Determine the Risk of Falls for Patients with Neuromuscular Disorders ENROLLING_BY_INVITATION false NCT00056810 Assessment of Chronic Guillain-Barre Syndrome Improvement With Use of 4-aminopyridine COMPLETED false NCT06032741 Prognosis in Patients With Guillain-Barre Syndrome UNKNOWN false NCT05494619 A Study to Evaluate the Efficacy, Safety, Pharma

In [0]:
# Convert Google Search Results JSON Data to Pandas DataFrame for Analysis


import pandas as pd
print(f"Length of fetched_results_raw: {len(fetched_results_raw)}")


try:
    serpapi_info = []

    for result in fetched_results_raw:
        serpapi_info.append({
            'Position': result.get('position', ''),
            'Title': result.get('title', ''),
            'Link': result.get('link', ''),
            'Snippet': result.get('snippet', ''),
            'Source': result.get('source', ''),
            'Displayed Link': result.get('displayed_link', ''),
            'Sitelinks': str(result.get('sitelinks', ''))  
        })

    df_serpapi = pd.DataFrame(serpapi_info)
    display(df_serpapi.head(100))

except NameError:
    print("Variable 'fetched_results_raw' not found. Please run the fetching cell first.")




Length of fetched_results_raw: 59


Position Title Link Snippet Source Displayed Link Sitelinks 1 Guillain-Barre syndrome - Symptoms and causes https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793 Guillain-Barre syndrome often begins with tingling and weakness starting in the feet and legs and spreading to the upper body and arms. Some ... Mayo Clinic https://www.mayoclinic.org › ... 2 Guillain-Barré Syndrome (GBS): Symptoms & Treatment https://my.clevelandclinic.org/health/diseases/15838-guillain-barre-syndrome Guillain-Barré syndrome is a rare autoimmune condition in which your immune system attacks your peripheral nerves. It causes numbness and muscle weakness. Cleveland Clinic https://my.clevelandclinic.org › ... 3 Guillain-Barré Syndrome https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome Guillain-Barré syndrome (GBS) (pronounced Ghee-yan Bah-ray) is a rare neurological disorder in which a person's immune system mistakenly attacks ... National Institute of Neurological Disorders and Stroke (.gov) https://www.ninds.nih.gov › ... {'inline': [{'title': 'What is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-is-guillain-barr-syndrome-'}, {'title': 'How is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-how-is-guillain-barr-syndrome-diagnosed-and-treated-'}, {'title': 'What are the latest updates on...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-are-the-latest-updates-on-guillain-barr-syndrome-'}]} 4 Guillain–Barré syndrome https://en.wikipedia.org/wiki/Guillain%E2%80%93Barr%C3%A9_syndrome Guillain–Barré syndrome (GBS) is a rapid-onset muscle weakness caused by the immune system damaging the peripheral nervous system. Wikipedia https://en.wikipedia.org › ... 5 Guillain–Barré syndrome https://www.who.int/news-room/fact-sheets/detail/guillain-barr%C3%A9-syndrome Guillain-Barré syndrome (GBS) is a rare condition in which a person's immune system attacks the peripheral nerves. World Health Organization (WHO) https://www.who.int › ... 6 Guillain-Barré Syndrome https://www.hopkinsmedicine.org/health/conditions-and-diseases/guillainbarr-syndrome Guillain-Barré syndrome (GBS) is a rare neurological disorder in which the body's immune system attacks the peripheral nervous system. Johns Hopkins Medicine https://www.hopkinsmedicine.org › ... 7 Guillain-Barre syndrome - Diagnosis and treatment https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/diagnosis-treatment/drc-20363006 This rare autoimmune condition affects the nerves, causing weakness and tingling in the arms and legs that quickly spreads throughout the ... Mayo Clinic https://www.mayoclinic.org › ... {'inline': [{'title': 'Symptoms and causes', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793'}, {'title': 'Doctors and departments', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/doctors-departments/ddc-20363037'}, {'title': 'Care at Mayo Clinic', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/care-at-mayo-clinic/mac-20363049'}]} 8 Guillain-Barre Syndrome https://www.physio-pedia.com/Guillain-Barre_Syndrome Guillain-Barré syndrome (GBS) is a condition characterised by the autoimmune destruction of the peripheral sensory system. Physiopedia https://www.physio-pedia.com › ... {'inline': [{'title': 'Case Study', 'link': 'https://www.physio-pedia.com/Case_Study:_Guillain-Barre_Syndrome_(Sub-Acute)'}, {'title': 'Acute Motor Axonal...', 'link': 'https://www.physio-pedia.com/Acute_Motor_Axonal_Neuropathy_(AMAN),_a_Variant_of_Guillain-Barre_Syndrome:_A_Case_Study'}, {'title': 'Edit', 'link': 'https://www.physio-pedia.com/Guillain-Barre_Syndrome?veaction=edit'}]} 9 Guillain-Barre Syndrome: Practice Essentials, Background, ..

**Defining PySpark Schemas and Creating Separate DataFrames for Clinical Trials and Google Search Results**

In [0]:
# Defining PySpark Schema and Creating Clinical Trials DataFrame
from pyspark.sql.types import *

clinical_schema = StructType([
    StructField("nctId", StringType(), True),
    StructField("briefTitle", StringType(), True),
    StructField("overallStatus", StringType(), True),
    StructField("hasResults", BooleanType(), True)
])

df_clinical_spark = spark.createDataFrame(trials_list, schema=clinical_schema)
display(df_clinical_spark)


nctId briefTitle overallStatus hasResults NCT05324176 Diaphragm Thickness by Ultrasonography in Neurological Disorders COMPLETED false NCT01655394 Change of Nerve Conduction Properties in IVIg Dependent Neuropathies UNKNOWN false NCT00271791 Prednisone Treatment for Vestibular Neuronitis COMPLETED false NCT06822231 High-Tech Rehabilitation Pathway for Acute Adult Neuromuscular Diseases - Fit4MedRob-Acute MND Project NOT_YET_RECRUITING false NCT06885762 Efgartigimod for the Treatment of Guillain-Barré Syndrome NOT_YET_RECRUITING false NCT05095844 National Vaccine Adverse Event Reporting Survey and Etiology UNKNOWN false NCT05701189 Evaluating Efgartigimod in Patients with Guillain-Barré Syndrome RECRUITING false NCT04752566 A Study to Evaluate the Efficacy and Safety of Eculizumab in Guillain-Barré Syndrome COMPLETED true NCT04829526 Firm Observational Clinical Unicenter Study on Guillain Barré Syndrome RECRUITING false NCT02493725 JET-GBS - Japanese Eculizumab Trial for GBS COMPLETED false NCT04166357 Early Prediction of Respiratory and Autonomic Complications of GBS Using Neuromuscular Ultrasound UNKNOWN false NCT05940922 RWE-based Treatment Patterns and Outcomes in CIDP NOT_YET_RECRUITING false NCT05212792 Genomics and COVID-19 Vaccine Adverse Events RECRUITING false NCT04303962 Efficacy of Intravenous Gamma Globulin on Guillain-Barre Syndrome UNKNOWN false NCT03943589 A Study of Imlifidase in Patients With Guillain-Barré Syndrome COMPLETED true NCT05104762 IVIG Versus Plasmapheresis in the Treatment of Guillian Barrie Syndrome Patients COMPLETED false NCT06615622 Our Study Aims to Determine If Nerve Alterations in Acute GBS and CIDP Detectable by Ultrasound Match Electrodiagnostic Findings and If This Method Aids Early Diagnosis, Predict Their Outcomes and Differentiate Between Axonal and Demyelinating Subtypes. RECRUITING false NCT05284786 Ultrasonography and Electrophysiology in GBS UNKNOWN false NCT01005524 Brain Computer Interface for Communication in ICU: a Feasibility Study COMPLETED false NCT06939634 Evaluation of Usability and Safety of the System Atalante in Patients With High Paraplegia and Tetraplegia COMPLETED false NCT04035135 A Clinical Study of ANX005 and IVIG in Subjects With Guillain Barré Syndrome (GBS) COMPLETED false NCT00167622 Early Mechanical Ventilation for Guillain Barré Syndrome COMPLETED false NCT01582763 International Guillain-Barré Syndrome Outcome Study COMPLETED false NCT03801135 Changes in Haemostasis After Therapeutic Plasmapheresis With Citrate Anticoagulation UNKNOWN false NCT03941184 Spontaneous Coronary Artery Dissection (SCAD) and Autoimmunity COMPLETED false NCT04092140 Neuromuscular Ultrasound in Correlation to Neurophysiological Findings UNKNOWN false NCT03268304 Feasibility of Two New Software Modules for the Rehabilitation of Patients With Neuromuscular Upper Limb Impairments WITHDRAWN false NCT00411216 Recovery of Visual Acuity in People With Vestibular Deficits COMPLETED true NCT01469858 Perception and Multisensory Integration in Neurological Patients Using fMRI UNKNOWN false NCT06167239 Ventilator Trigger Sensitivity Adjustment Versus Threshold Inspiratory Muscle Training on Arterial Blood Gases RECRUITING false NCT04674163 Expression Profile of ERK5 and PKM2 Kinases in Neuroinflammatory Diseases. WITHDRAWN false NCT05630313 Gene Sequencing as a Strategy for Identifying Genetic Factors Associated With Serious Adverse Events After Covid-19 Vaccines in Use in Brazil RECRUITING false NCT02459808 Gastrointestinal Transit Time in Patients With Guillain-Barré Syndrome WITHDRAWN false NCT06605612 Development and Validation of the FBIndex to Determine the Risk of Falls for Patients with Neuromuscular Disorders ENROLLING_BY_INVITATION false NCT00056810 Assessment of Chronic Guillain-Barre Syndrome Improvement With Use of 4-aminopyridine COMPLETED false NCT06032741 Prognosis in Patients With Guillain-Barre Syndrome UNKNOWN false NCT05494619 A Study to Evaluate the Efficacy, Safety, Pharma

In [0]:
# Defining PySpark Schema and Creating Google Search Results DataFrame

from pyspark.sql.types import *

serpapi_schema = StructType([
    StructField("Position", IntegerType(), True),
    StructField("Title", StringType(), True),
    StructField("Link", StringType(), True),
    StructField("Snippet", StringType(), True),
    StructField("Source", StringType(), True),
    StructField("Displayed Link", StringType(), True),
    StructField("Sitelinks", StringType(), True)
])

df_serpapi_spark = spark.createDataFrame(df_serpapi, schema=serpapi_schema)
display(df_serpapi_spark)


Position Title Link Snippet Source Displayed Link Sitelinks 1 Guillain-Barre syndrome - Symptoms and causes https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793 Guillain-Barre syndrome often begins with tingling and weakness starting in the feet and legs and spreading to the upper body and arms. Some ... Mayo Clinic https://www.mayoclinic.org › ... 2 Guillain-Barré Syndrome (GBS): Symptoms & Treatment https://my.clevelandclinic.org/health/diseases/15838-guillain-barre-syndrome Guillain-Barré syndrome is a rare autoimmune condition in which your immune system attacks your peripheral nerves. It causes numbness and muscle weakness. Cleveland Clinic https://my.clevelandclinic.org › ... 3 Guillain-Barré Syndrome https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome Guillain-Barré syndrome (GBS) (pronounced Ghee-yan Bah-ray) is a rare neurological disorder in which a person's immune system mistakenly attacks ... National Institute of Neurological Disorders and Stroke (.gov) https://www.ninds.nih.gov › ... {'inline': [{'title': 'What is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-is-guillain-barr-syndrome-'}, {'title': 'How is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-how-is-guillain-barr-syndrome-diagnosed-and-treated-'}, {'title': 'What are the latest updates on...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-are-the-latest-updates-on-guillain-barr-syndrome-'}]} 4 Guillain–Barré syndrome https://en.wikipedia.org/wiki/Guillain%E2%80%93Barr%C3%A9_syndrome Guillain–Barré syndrome (GBS) is a rapid-onset muscle weakness caused by the immune system damaging the peripheral nervous system. Wikipedia https://en.wikipedia.org › ... 5 Guillain–Barré syndrome https://www.who.int/news-room/fact-sheets/detail/guillain-barr%C3%A9-syndrome Guillain-Barré syndrome (GBS) is a rare condition in which a person's immune system attacks the peripheral nerves. World Health Organization (WHO) https://www.who.int › ... 6 Guillain-Barré Syndrome https://www.hopkinsmedicine.org/health/conditions-and-diseases/guillainbarr-syndrome Guillain-Barré syndrome (GBS) is a rare neurological disorder in which the body's immune system attacks the peripheral nervous system. Johns Hopkins Medicine https://www.hopkinsmedicine.org › ... 7 Guillain-Barre syndrome - Diagnosis and treatment https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/diagnosis-treatment/drc-20363006 This rare autoimmune condition affects the nerves, causing weakness and tingling in the arms and legs that quickly spreads throughout the ... Mayo Clinic https://www.mayoclinic.org › ... {'inline': [{'title': 'Symptoms and causes', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793'}, {'title': 'Doctors and departments', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/doctors-departments/ddc-20363037'}, {'title': 'Care at Mayo Clinic', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/care-at-mayo-clinic/mac-20363049'}]} 8 Guillain-Barre Syndrome https://www.physio-pedia.com/Guillain-Barre_Syndrome Guillain-Barré syndrome (GBS) is a condition characterised by the autoimmune destruction of the peripheral sensory system. Physiopedia https://www.physio-pedia.com › ... {'inline': [{'title': 'Case Study', 'link': 'https://www.physio-pedia.com/Case_Study:_Guillain-Barre_Syndrome_(Sub-Acute)'}, {'title': 'Acute Motor Axonal...', 'link': 'https://www.physio-pedia.com/Acute_Motor_Axonal_Neuropathy_(AMAN),_a_Variant_of_Guillain-Barre_Syndrome:_A_Case_Study'}, {'title': 'Edit', 'link': 'https://www.physio-pedia.com/Guillain-Barre_Syndrome?veaction=edit'}]} 9 Guillain-Barre Syndrome: Practice Essentials, Background, ..

**Cleaning and Preparation of Clinical Trials Data**

In [0]:
# Clinical Trials Data Summary Statistics
display(df_clinical_spark.describe())

summary nctId briefTitle overallStatus count 76 76 76 mean null null null stddev null null null min NCT00004833 A Clinical Study of ANX005 and IVIG in Subjects With Guillain Barré Syndrome (GBS) ACTIVE_NOT_RECRUITING max NCT06940908 sCD163 as a Potential Biomarker in Guillain- Barré Syndrome WITHDRAWN

In [0]:
# Print Schema of Clinical Trials DataFrame
df_clinical_spark.printSchema()

root
 |-- nctId: string (nullable = true)
 |-- briefTitle: string (nullable = true)
 |-- overallStatus: string (nullable = true)
 |-- hasResults: boolean (nullable = true)



In [0]:
# Check for Null Values in Clinical Trials DataFrame Columns

import pyspark.sql.functions as p
df_clinical_spark.select([
    p.count(p.when(p.col(c).isNull(), c)).alias(c) for c in df_clinical_spark.columns
]).display()


nctId briefTitle overallStatus hasResults 0 0 0 0

In [0]:
# Check for Duplicate Rows in Clinical Trials DataFrame

print("Duplicates in Clinical Data:")
df_clinical_spark.count() - df_clinical_spark.dropDuplicates().count()


Duplicates in Clinical Data:
Out[10]: 0

**Cleaning and Preparation of Google Search Results Data**

In [0]:
#Statistics Summary of Google Search Results Data
display(df_serpapi_spark.describe())

summary Position Title Link Snippet Source Displayed Link Sitelinks count 59 59 59 59 59 59 59 mean 5.423728813559322 null null null null null null stddev 2.8599697962626767 null null null null null null min 1 Current treatment practice of Guillain-Barré syndrome https://emedicine.medscape.com/article/315632-overview A rare condition in which the body's immune system attacks the nerves located outside the brain and spinal cord. Symptoms of Guillain-Barré syndrome include ... American Brain Foundation https://emedicine.medscape.com › ... max 10 क्या है Guillain-Barre Syndrome? महाराष्ट्र में 6 की मौत ... https://zerotofinals.com/medicine/neurology/guillainbarre/ गिलियन-बैरे सिंड्रोम (जीबीएस) एक ऑटोइम्यून बीमारी है. यह मरीजों के इम्यून सिस्टम को प्रभावित करता है. जीबीएस मरीजों के स्वस्थ्य नसों पर हमला करता है. जिससे नसें कमजोर हो जाती हैं. अगर समय पर इसका इलाज ... adhikarilifeline.com https://zerotofinals.com › ... {'inline': [{'title': 'What is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-is-guillain-barr-syndrome-'}, {'title': 'How is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-how-is-guillain-barr-syndrome-diagnosed-and-treated-'}, {'title': 'What are the latest updates on...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-are-the-latest-updates-on-guillain-barr-syndrome-'}]}

In [0]:
# Print Schema of Google Search Results DataFrame
df_serpapi_spark.printSchema()


root
 |-- Position: integer (nullable = true)
 |-- Title: string (nullable = true)
 |-- Link: string (nullable = true)
 |-- Snippet: string (nullable = true)
 |-- Source: string (nullable = true)
 |-- Displayed Link: string (nullable = true)
 |-- Sitelinks: string (nullable = true)



In [0]:
# Check for Null Values in Google Search Results DataFrame

import pyspark.sql.functions as p

df_serpapi_spark.select([
    p.count(p.when(p.col(c).isNull(), c)).alias(c) for c in df_serpapi_spark.columns
]).display()


Position Title Link Snippet Source Displayed Link Sitelinks 0 0 0 0 0 0 0

In [0]:
# Check for Duplicates in Google Search Results DataFrame

print("Duplicates in SERP API Data:")
df_serpapi_spark.count() - df_serpapi_spark.dropDuplicates().count()


Duplicates in SERP API Data:
Out[14]: 0

In [0]:
# Identify and Display Duplicate Rows in Google Search Results DataFrame

from pyspark.sql import functions as p

# Group by all columns and count occurrences
duplicate_rows = (
    df_serpapi_spark
    .groupBy(df_serpapi_spark.columns)
    .count()
    .filter("count > 1")
)

# Show the duplicate rows
duplicate_rows.display(truncate=False)


Position Title Link Snippet Source Displayed Link Sitelinks count

In [0]:
# Remove Duplicate Rows from Google Search Results DataFrame and Report Count

before = df_serpapi_spark.count()
df_serpapi_spark = df_serpapi_spark.dropDuplicates()
after = df_serpapi_spark.count()

print(f"Removed {before - after} duplicate rows")


Removed 0 duplicate rows


In [0]:
# Fill Missing Values in 'Sitelinks' Column of Google Search Results DataFrame

df_serpapi_spark = df_serpapi_spark.na.fill({'Sitelinks': 'No Sitelinks'})

In [0]:
# Identify Duplicate Rows in Google Search Results DataFrame

df_serpapi_spark.groupBy(df_serpapi_spark.columns) \
    .count().filter("count > 1").display()


Position Title Link Snippet Source Displayed Link Sitelinks count

In [0]:
# Count Null Values in Each Column of Google Search Results DataFrame

import pyspark.sql.functions as p

df_serpapi_spark.select([
    p.count(p.when(p.col(c).isNull(), c)).alias(c) for c in df_serpapi_spark.columns
]).display()


Position Title Link Snippet Source Displayed Link Sitelinks 0 0 0 0 0 0 0

In [0]:
# Clean column names for df_serpapi_spark
df_serpapi_spark = df_serpapi_spark.withColumnRenamed("Displayed Link", "Displayed_Link")

In [0]:
# Display Cleaned Google Search Results DataFrame

display(df_serpapi_spark)

Position Title Link Snippet Source Displayed_Link Sitelinks 6 Guillain-Barré Syndrome https://www.hopkinsmedicine.org/health/conditions-and-diseases/guillainbarr-syndrome Guillain-Barré syndrome (GBS) is a rare neurological disorder in which the body's immune system attacks the peripheral nervous system. Johns Hopkins Medicine https://www.hopkinsmedicine.org › ... 5 Guillain–Barré syndrome https://www.who.int/news-room/fact-sheets/detail/guillain-barr%C3%A9-syndrome Guillain-Barré syndrome (GBS) is a rare condition in which a person's immune system attacks the peripheral nerves. World Health Organization (WHO) https://www.who.int › ... 2 Guillain-Barré Syndrome (GBS): Symptoms & Treatment https://my.clevelandclinic.org/health/diseases/15838-guillain-barre-syndrome Guillain-Barré syndrome is a rare autoimmune condition in which your immune system attacks your peripheral nerves. It causes numbness and muscle weakness. Cleveland Clinic https://my.clevelandclinic.org › ... 1 Guillain-Barre syndrome - Symptoms and causes https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793 Guillain-Barre syndrome often begins with tingling and weakness starting in the feet and legs and spreading to the upper body and arms. Some ... Mayo Clinic https://www.mayoclinic.org › ... 7 Guillain-Barre syndrome - Diagnosis and treatment https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/diagnosis-treatment/drc-20363006 This rare autoimmune condition affects the nerves, causing weakness and tingling in the arms and legs that quickly spreads throughout the ... Mayo Clinic https://www.mayoclinic.org › ... {'inline': [{'title': 'Symptoms and causes', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793'}, {'title': 'Doctors and departments', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/doctors-departments/ddc-20363037'}, {'title': 'Care at Mayo Clinic', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/care-at-mayo-clinic/mac-20363049'}]} 4 Guillain–Barré syndrome https://en.wikipedia.org/wiki/Guillain%E2%80%93Barr%C3%A9_syndrome Guillain–Barré syndrome (GBS) is a rapid-onset muscle weakness caused by the immune system damaging the peripheral nervous system. Wikipedia https://en.wikipedia.org › ... 3 Guillain-Barré Syndrome https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome Guillain-Barré syndrome (GBS) (pronounced Ghee-yan Bah-ray) is a rare neurological disorder in which a person's immune system mistakenly attacks ... National Institute of Neurological Disorders and Stroke (.gov) https://www.ninds.nih.gov › ... {'inline': [{'title': 'What is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-is-guillain-barr-syndrome-'}, {'title': 'How is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-how-is-guillain-barr-syndrome-diagnosed-and-treated-'}, {'title': 'What are the latest updates on...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-are-the-latest-updates-on-guillain-barr-syndrome-'}]} 8 Guillain-Barre Syndrome https://www.physio-pedia.com/Guillain-Barre_Syndrome Guillain-Barré syndrome (GBS) is a condition characterised by the autoimmune destruction of the peripheral sensory system. Physiopedia https://www.physio-pedia.com › ... {'inline': [{'title': 'Case Study', 'link': 'https://www.physio-pedia.com/Case_Study:_Guillain-Barre_Syndrome_(Sub-Acute)'}, {'title': 'Acute Motor Axonal...', 'link': 'https://www.physio-pedia.com/Acute_Motor_Axonal_Neuropathy_(AMAN),_a_Variant_of_Guillain-Barre_Syndrome:_A_Case_Study'}, {'title': 'Edit', 'link': 'https://www.physio-pedia.com/Guillain-Barre_Syndrome?veaction=edit'}]} 1 Guillain-Barré syndrome Information - Mount Sinai https://ww

**Exploratory Data Analysis (EDA) of Clinical Trials and Google Search Data**

**Distribution of Study Status in Clinical Trials (overallStatus)**

In [0]:
# Understand the distribution of studies by status — e.g., completed, recruiting, withdrawn, etc.

df_clinical_spark.groupBy("overallStatus") \
    .count() \
    .orderBy("count", ascending=False) \
    .display(truncate=False)


overallStatus count COMPLETED 31 UNKNOWN 16 RECRUITING 12 NOT_YET_RECRUITING 7 WITHDRAWN 5 ENROLLING_BY_INVITATION 2 TERMINATED 2 ACTIVE_NOT_RECRUITING 1

**Percentage of studies with results vs without results**

In [0]:
# Quick insight into how many studies have published results — and whether most clinical trials on GBS report their findings
from pyspark.sql.functions import col

total_studies = df_clinical_spark.count()
results_group = df_clinical_spark.groupBy("hasResults").count()

results_group.withColumn(
    "percentage", 
    (col("count") / total_studies * 100).cast("decimal(5,2)")
).display()


hasResults count percentage true 4 5.26 false 72 94.74

**Filter studies with results for focused analysis**

In [0]:
# Filter and preview studies that have published results for further analysis
df_with_results = df_clinical_spark.filter(col("hasResults") == True)
print(f"Studies with results: {df_with_results.count()}")
df_with_results.display(5, truncate=False)



Studies with results: 4


nctId briefTitle overallStatus hasResults NCT04752566 A Study to Evaluate the Efficacy and Safety of Eculizumab in Guillain-Barré Syndrome COMPLETED true NCT03943589 A Study of Imlifidase in Patients With Guillain-Barré Syndrome COMPLETED true NCT00411216 Recovery of Visual Acuity in People With Vestibular Deficits COMPLETED true NCT04053452 Peripheral Nerve Ultrasound for Diagnosis and Prognosis of Guillain-Barre Syndrome TERMINATED true

**Exploratory Data Analysis of Google Search Results**

**Source-wise Result Count with Best Position**

In [0]:
#Source-wise count of Google Search results and their best positions

from pyspark.sql.functions import col, count, min

df_serpapi_spark.groupBy("Source") \
    .agg(
        count("*").alias("No_of_Results"),
        min("Position").alias("Best_Position")  # Lower means higher rank
    ) \
    .orderBy(col("No_of_Results").desc(), col("Best_Position")) \
    .display(truncate=False)


Source No_of_Results Best_Position Mayo Clinic 2 1 BLK-Max Hospital 2 1 Centers for Disease Control and Prevention | CDC (.gov) 2 2 WebMD 2 3 Nature 2 3 ScienceDirect.com 2 5 Zero To Finals 2 6 Mount Sinai 1 1 Healthline 1 1 National Organization for Rare Disorders 1 1 Wolters Kluwer 1 2 Children’s Health 1 2 Cleveland Clinic 1 2 Yale Medicine 1 2 National Institutes of Health (NIH) (.gov) 1 2 National Cancer Institute (.gov) 1 3 Neurology® Journals 1 3 National Institute of Neurological Disorders and Stroke (.gov) 1 3 Jefferson Health 1 3 Vax-Before-Travel 1 4 Wikipedia 1 4 Osmosis 1 4 The New England Journal of Medicine 1 4 American Brain Foundation 1 5 World Health Organization (WHO) 1 5 Life in the Fast Lane • LITFL 1 5 Indian Journal of Neurosciences 1 5 Johns Hopkins Medicine 1 6 adhikarilifeline.com 1 6 Lippincott 1 6 Saint Luke's Health System Kansas City 1 6 Pulmonology Advisor 1 6 NewYork-Presbyterian 1 7 KidsHealth 1 7 PACE Hospitals 1 7 MedlinePlus (.gov) 1 7 Wiley Online Library 1 7 GPnotebook 1 8 Boston Children's Hospital 1 8 MSD Manuals 1 8 UTHealth Houston 1 8 BrainFacts 1 8 Physiopedia 1 8 The Lancet 1 9 Medcomic 1 9 Medscape 1 9 Medical News Today 1 9 Mayo Clinic Proceedings 1 9 GBS/CIDP Foundation International 1 10 ChoosePT 1 10 University of Rochester Medical Center 1 10 Prabhat Khabar 1 10

**Which Titles appear most frequently in the top 3 positions, and which sources do they come from**

In [0]:
# Most frequent Titles in top 3 positions along with their Sources

from pyspark.sql.functions import col, count, collect_set

# Filter top 3 positions
top_positions_df = df_serpapi_spark.filter(col("Position") <= 3)

# Group by Title and collect the Sources
top_titles_with_sources = top_positions_df.groupBy("Title") \
    .agg(
        count("*").alias("Count"),
        collect_set("Source").alias("Top_Sources")
    ) \
    .orderBy(col("Count").desc())

top_titles_with_sources.display(truncate=False)



Title Count Top_Sources Guillain-Barré Syndrome 2 List(National Institute of Neurological Disorders and Stroke (.gov), Yale Medicine) Guillain-Barre Syndrome: Symptoms, Causes & Treatment 2 List(BLK-Max Hospital) Guillain-Barre Syndrome - StatPearls 1 List(National Institutes of Health (NIH) (.gov)) Guillain-Barre Syndrome (GBS) 1 List(Jefferson Health) Diagnosis and management of Guillain–Barré syndrome ... 1 List(Nature) Guillain-Barré syndrome in adults: Pathogenesis, clinical ... 1 List(Wolters Kluwer) Guillain-Barré Syndrome (GBS) and Vaccines 1 List(Centers for Disease Control and Prevention | CDC (.gov)) Pediatric Guillain-Barré syndrome (GBS) - Dallas 1 List(Children’s Health) Guillain-Barré syndrome (GBS) 1 List(Healthline) Guillain-Barré Syndrome: Causes, Symptoms, ... - WebMD 1 List(WebMD) Definition of Guillain-Barré syndrome 1 List(National Cancer Institute (.gov)) Guillain-Barré syndrome Information - Mount Sinai 1 List(Mount Sinai) Guillain-Barre syndrome - Symptoms and causes 1 List(Mayo Clinic) Current treatment practice of Guillain-Barré syndrome 1 List(Neurology® Journals) Guillain-Barré Syndrome (GBS): Symptoms & Treatment 1 List(Cleveland Clinic) Guillain-Barré Syndrome - Symptoms, Causes, Treatment 1 List(National Organization for Rare Disorders)

**Topic Categorization of Google Search Snippets**

In [0]:
# Categorize search result snippets into medical topics like Symptoms, Causes, Treatment, etc. and count occurrences

from pyspark.sql.functions import when, col

df_tagged = df_serpapi_spark.withColumn("Topic",
    when(col("Snippet").rlike("(?i)symptom"), "Symptoms")
    .when(col("Snippet").rlike("(?i)cause|trigger"), "Causes")
    .when(col("Snippet").rlike("(?i)treat|therapy|manage"), "Treatment")
    .when(col("Snippet").rlike("(?i)recover|rehab"), "Recovery")
    .when(col("Snippet").rlike("(?i)diagnos"), "Diagnosis")
    .otherwise("Other")
)

df_tagged.groupBy("Topic").count().orderBy("count", ascending=False).display()


Topic count Other 36 Causes 10 Symptoms 8 Diagnosis 2 Recovery 2 Treatment 1

**Saving Clinical Trials and Google Search Results DataFrames as Spark Tables**

In [0]:
#Saving Clinical Trials DataFrame as a Spark Table
df_clinical_spark.write.mode("overwrite").saveAsTable("final_clinical_data")


In [0]:
%sql
select * from final_clinical_data

nctId briefTitle overallStatus hasResults NCT01024088 Prospective Study on Swallowing/Breathing Interactions in Severe Guillain Barre Syndrome. COMPLETED false NCT05461898 RehabGBs: Rehabilitation in People With Guillain-Barré Syndrome RECRUITING false NCT02780570 Small Volume Plasma Exchange (SVPE) for Guillain-Barré Syndrome (GBS) Patients COMPLETED false NCT02722070 Processing Integration in Neurological Patients Using fMRI UNKNOWN false NCT04927598 Predictors and Prognostic Factors of Gullian Barrie Syndrome Outcome COMPLETED false NCT03773328 A Clinical Trial of CK0801 (a New Drug) In Patients With Treatment-Resistant Guillain-Barré Syndrome (GBS) WITHDRAWN false NCT03840928 PatientSpot Formerly Known as ArthritisPower RECRUITING false NCT02582853 sCD163 as a Potential Biomarker in Guillain- Barré Syndrome UNKNOWN false NCT06200454 Predictive Value of Neuromuscular Ultrasound of Cranial Nerves in Guillain-Barré Syndrome NOT_YET_RECRUITING false NCT00004833 Randomized Study of Plasmapheresis or Human Immunoglobulin Infusion in Childhood Guillain-Barre Syndrome TERMINATED false NCT00575653 Safety Study of GBS Following Menactra Meningococcal Vaccination COMPLETED false NCT06041451 Early and Late Prognosis in Patients With Guillain-Barre Syndrome UNKNOWN false NCT06620198 Upper Limb Robot-Assisted Therapy in Patients with Guillain-Barré Syndrome ACTIVE_NOT_RECRUITING false NCT05494619 A Study to Evaluate the Efficacy, Safety, Pharmacokinetics, and Pharmacodynamics of Crovalimab in Participants With Guillain-Barré Syndrome (GBS) WITHDRAWN false NCT05114941 Comparison of the Efficacy and Safety of Immunoadsorption and Intravenous Immunoglobulin for Guillain-Barre Syndrome NOT_YET_RECRUITING false NCT03206541 Neurologic Manifestations of the Arbovirus Infection in Colombia COMPLETED false NCT06334796 Artificial Intelligence-powered Virtual Assistant for Emergency Triage in Neurology COMPLETED false NCT05491096 Effects of Proprioceptive Neuromuscular Training on Sensory and Motor Function in Guillain Barre Syndrome COMPLETED false NCT04871035 Immunoadsorption Versus Plasma Exchange for Treatment of Guillain-Barré Syndrome (GBS) RECRUITING false NCT06612242 Early vs. Late Tracheostomy in Patients With Guillain -Barre Syndrome RECRUITING false NCT02883270 Effects of Robotic-assisted Gait Training In Non-Ambulatory Patients After Guillain-Barré Syndrome COMPLETED false NCT00173199 The Changes of Cytokines in Guillain Barré Syndrome: the Correlation With Clinical Manifestations and Skin Innervation UNKNOWN false NCT00411216 Recovery of Visual Acuity in People With Vestibular Deficits COMPLETED true NCT01469858 Perception and Multisensory Integration in Neurological Patients Using fMRI UNKNOWN false NCT06167239 Ventilator Trigger Sensitivity Adjustment Versus Threshold Inspiratory Muscle Training on Arterial Blood Gases RECRUITING false NCT04674163 Expression Profile of ERK5 and PKM2 Kinases in Neuroinflammatory Diseases. WITHDRAWN false NCT05630313 Gene Sequencing as a Strategy for Identifying Genetic Factors Associated With Serious Adverse Events After Covid-19 Vaccines in Use in Brazil RECRUITING false NCT02459808 Gastrointestinal Transit Time in Patients With Guillain-Barré Syndrome WITHDRAWN false NCT06605612 Development and Validation of the FBIndex to Determine the Risk of Falls for Patients with Neuromuscular Disorders ENROLLING_BY_INVITATION false NCT00056810 Assessment of Chronic Guillain-Barre Syndrome Improvement With Use of 4-aminopyridine COMPLETED false NCT06032741 Prognosis in Patients With Guillain-Barre Syndrome UNKNOWN false NCT02493725 JET-GBS - Japanese Eculizumab Trial for GBS COMPLETED false NCT04166357 Early Prediction of Respiratory and Autonomic Complications of GBS Using Neuromuscular Ultrasound UNKNOWN false NCT05940922 RWE-based Treatment Patterns and Outcomes in CIDP NOT_YET_RECRUITING false NCT05212792 Genomics and COVID-19 Vaccine Adverse Events RECRUITING false NCT04303962 Efficacy of Intravenous Gam

In [0]:
#Saving Google Search Results DataFrame as a Spark Table
df_serpapi_spark.write.mode("overwrite").saveAsTable("final_serpapi_data")


In [0]:
%sql
select * from final_serpapi_data

Position Title Link Snippet Source Displayed_Link Sitelinks 6 Guillain-Barré Syndrome https://www.hopkinsmedicine.org/health/conditions-and-diseases/guillainbarr-syndrome Guillain-Barré syndrome (GBS) is a rare neurological disorder in which the body's immune system attacks the peripheral nervous system. Johns Hopkins Medicine https://www.hopkinsmedicine.org › ... 5 Guillain–Barré syndrome https://www.who.int/news-room/fact-sheets/detail/guillain-barr%C3%A9-syndrome Guillain-Barré syndrome (GBS) is a rare condition in which a person's immune system attacks the peripheral nerves. World Health Organization (WHO) https://www.who.int › ... 2 Guillain-Barré Syndrome (GBS): Symptoms & Treatment https://my.clevelandclinic.org/health/diseases/15838-guillain-barre-syndrome Guillain-Barré syndrome is a rare autoimmune condition in which your immune system attacks your peripheral nerves. It causes numbness and muscle weakness. Cleveland Clinic https://my.clevelandclinic.org › ... 1 Guillain-Barre syndrome - Symptoms and causes https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793 Guillain-Barre syndrome often begins with tingling and weakness starting in the feet and legs and spreading to the upper body and arms. Some ... Mayo Clinic https://www.mayoclinic.org › ... 7 Guillain-Barre syndrome - Diagnosis and treatment https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/diagnosis-treatment/drc-20363006 This rare autoimmune condition affects the nerves, causing weakness and tingling in the arms and legs that quickly spreads throughout the ... Mayo Clinic https://www.mayoclinic.org › ... {'inline': [{'title': 'Symptoms and causes', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/symptoms-causes/syc-20362793'}, {'title': 'Doctors and departments', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/doctors-departments/ddc-20363037'}, {'title': 'Care at Mayo Clinic', 'link': 'https://www.mayoclinic.org/diseases-conditions/guillain-barre-syndrome/care-at-mayo-clinic/mac-20363049'}]} 4 Guillain–Barré syndrome https://en.wikipedia.org/wiki/Guillain%E2%80%93Barr%C3%A9_syndrome Guillain–Barré syndrome (GBS) is a rapid-onset muscle weakness caused by the immune system damaging the peripheral nervous system. Wikipedia https://en.wikipedia.org › ... 3 Guillain-Barré Syndrome https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome Guillain-Barré syndrome (GBS) (pronounced Ghee-yan Bah-ray) is a rare neurological disorder in which a person's immune system mistakenly attacks ... National Institute of Neurological Disorders and Stroke (.gov) https://www.ninds.nih.gov › ... {'inline': [{'title': 'What is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-is-guillain-barr-syndrome-'}, {'title': 'How is Guillain-Barré...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-how-is-guillain-barr-syndrome-diagnosed-and-treated-'}, {'title': 'What are the latest updates on...', 'link': 'https://www.ninds.nih.gov/health-information/disorders/guillain-barre-syndrome#toc-what-are-the-latest-updates-on-guillain-barr-syndrome-'}]} 8 Guillain-Barre Syndrome https://www.physio-pedia.com/Guillain-Barre_Syndrome Guillain-Barré syndrome (GBS) is a condition characterised by the autoimmune destruction of the peripheral sensory system. Physiopedia https://www.physio-pedia.com › ... {'inline': [{'title': 'Case Study', 'link': 'https://www.physio-pedia.com/Case_Study:_Guillain-Barre_Syndrome_(Sub-Acute)'}, {'title': 'Acute Motor Axonal...', 'link': 'https://www.physio-pedia.com/Acute_Motor_Axonal_Neuropathy_(AMAN),_a_Variant_of_Guillain-Barre_Syndrome:_A_Case_Study'}, {'title': 'Edit', 'link': 'https://www.physio-pedia.com/Guillain-Barre_Syndrome?veaction=edit'}]} 1 Guillain-Barré syndrome Information - Mount Sinai https://ww

**Derived Results Summary**

**Clinical Trials Data Insights:**

- The majority of clinical trials related to Guillain-Barré Syndrome (GBS) are completed (31 studies), while a significant portion remains unknown or recruiting.

- Only about 5.26% of the studies have published results, indicating that most clinical trials on GBS have yet to report their findings.

- A filtered subset of studies with results was created for focused analysis, highlighting the available data for deeper examination.

**SERP API Search Results Insights:**

- The most frequent sources in Google search results include authoritative health organizations such as the Centers for Disease Control and Prevention (CDC), Mayo Clinic, and The Lancet, often appearing at top ranks.

- Titles related to Guillain-Barré Syndrome appearing in the top 3 search positions are mostly from reputable medical institutions, emphasizing the dominance of credible sources in search results.

- Search snippets are mainly focused on Symptoms, followed by Causes and Treatment, suggesting public interest and information availability centers on understanding and managing the condition.